diff --git a/apriltag_ros/include/AprilTagNode.hpp b/apriltag_ros/include/AprilTagNode.hpp
index 8abbbcc..fa21b80 100644
--- a/apriltag_ros/include/AprilTagNode.hpp
+++ b/apriltag_ros/include/AprilTagNode.hpp
@@ -85,7 +85,7 @@ private:
   std::unordered_map<int, double> tag_sizes;
   std::unique_ptr<tf2_ros::TransformBroadcaster> tf_broadcaster_;

-  bool remove_duplicates_ = true;
+  bool remove_duplicates_ = false;
   bool z_up;
   // function pointer for tag family creation / destruction
   static const std::map<std::string, apriltag_family_t * (*)(void)> tag_create;
