diff --git a/turtlebot3_gazebo/models/turtlebot3_waffle_pi/model.sdf b/turtlebot3_gazebo/models/turtlebot3_waffle_pi/model.sdf
index f7658dd..5f6673a 100644
--- a/turtlebot3_gazebo/models/turtlebot3_waffle_pi/model.sdf
+++ b/turtlebot3_gazebo/models/turtlebot3_waffle_pi/model.sdf
@@ -131,7 +131,7 @@
 
       <sensor name="hls_lfcd_lds" type="ray">
         <always_on>true</always_on>
-        <visualize>true</visualize>
+        <visualize>false</visualize>
         <pose>-0.064 0 0.121 0 0 0</pose>
         <update_rate>5</update_rate>
         <ray>
@@ -145,7 +145,7 @@
           </scan>
           <range>
             <min>0.120000</min>
-            <max>3.5</max>
+            <max>12.0</max>
             <resolution>0.015000</resolution>
           </range>
           <noise>
